### PR TITLE
[GEOS-10567] Fix module modularity for OpenIdConnect and oauth2-core

### PR DESCRIPTION
**Pull request by ahmedababnehgeo**
[GEOS-10567](https://osgeo-org.atlassian.net/browse/GEOS-10567)

Refactor the OpenId Connect community module separating the `web` functionality away form the core functionality each in a separate module.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by ahmedababnehgeo** [comment]
@taba90 PR ready for review

**Review by taba90 — CHANGES_REQUESTED** [review]
The splitting of the module seem fine however, I guess that at least the gs-oauth2-core module should be splitted in the same way as well.
The openid-connect profile in the gs-web-app pom.xml has not been updated remember to add the renamed dependencies to it and remove the old one.
Finally the pom in /community/release folder should be updated as well to replace the old opend id dependency the new ones.

File: src/community/security/oauth2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    <version>2.22-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.geoserver.community.security</groupId>
+  <artifactId>gs-sec-oauth2</artifactId>
+  <packaging>pom</packaging>
+  <name>GeoServer OAuth2 Connect Security Module</name>
+
+  <modules>
+    <module>oauth2-core</module>
```
Comment (taba90) [inline src/community/security/oauth2/pom.xml:24]: missing the oauth2-web module

File: src/community/pom.xml [hunk 2]
```diff
@@ -218,7 +218,7 @@
         <module>gwc-sqlite</module>
         <module>geostyler</module>
         <module>security</module>
-        <module>security/oauth2-core</module>
```
Comment (taba90) [inline src/community/pom.xml:221]: missing the oauth2-web module addition

File: src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/pom.xml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  <name>GeoServer OAuth2 Connect Security Core Module - OpenId Connect</name>
+
+  <dependencies>
+
+    <dependency>
+      <groupId>org.geoserver</groupId>
+      <artifactId>gs-main</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
```
Comment (taba90) [inline src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/pom.xml:29]: missing the dependency for the oauth2-web

File: src/community/security/oauth2/oauth2-web/pom.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (taba90) [inline src/community/security/oauth2/oauth2-web/pom.xml:1]: We have the module correctly separated from the core one but is not added as a dependency where it is needed.

**Review by taba90 — CHANGES_REQUESTED** [review]
The web parte of the oauth2 core module (oauth2/oauth2-web) has not been added as dependency in all the needed places. I've commented in some of them but not all.
The oauth2-web dependency/module needs to be present:
- in the web-app pom.xml
- in the community/release pom.xml
- in the community pom.xml (here all the profile for oauth2 related implementation have still reference to the old oauth2-core module fix them as well)
- in the pom.xml of each oauth2 implementation (google, geonode, github, openid)

**Comment by ahmedababnehgeo** [comment]
@taba90 PR ready for review again

**Review by taba90 — APPROVED** [review]

**Comment by jodygarnett** [comment]
I am having trouble building this cleanly with -PcommunityRelease,oauth2-openid-connect:

```
[ERROR] [ERROR] Project 'org.geoserver.community.security:gs-sec-oauth2-core:2.21-90bb3fce62' is duplicated in the reactor @ 
[ERROR] Project 'org.geoserver.community.security:gs-sec-oauth2-core:2.21-90bb3fce62' is duplicated in the reactor -> [Help 1]
```

This is because the module structure is not consistent across these two profiles:

community/pom.xml profilecommunityRelease profile has:
````
        <module>security</module>
        <module>security/oauth2/oauth2-core</module>
        <module>security/oauth2/oauth2-web</module>
        <module>security/oauth2-google</module>
        <module>security/oauth2-github</module>
        <module>security/oauth2-geonode</module>
        <module>security/oauth2-openid-connect</module>
        <module>security/keycloak</module>
````

community/pom.xml profile oauth2-openid-connect:
````
        <module>security/oauth2/oauth2-core</module>
        <module>security/oauth2/oauth2-web</module>
        <module>security/oauth2-openid-connect</module>
````

community/security/pom.xml profile oauth2-openid-connect:
```
        <module>oauth2</module>
        <module>oauth2-openid-connect</module>
```

**Pull request merged**